fabric-dimensions-v1/src/main/java/net/fabricmc/fabric/api/dimension/FabricEntityTeleporter.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2016, 2017, 2018 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.dimension;
+
+import net.fabricmc.fabric.impl.entity.TeleportingServerPlayerEntity;
+import net.minecraft.client.network.packet.*;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.effect.StatusEffectInstance;
+import net.minecraft.server.PlayerManager;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.dimension.DimensionType;
+import net.minecraft.world.level.LevelProperties;
+
+/**
+ * Entity teleporter that ignores vanilla placement actions (such as nether portals being built).
+ */
+public class FabricEntityTeleporter {
+    public static void changeDimension(Entity entity, DimensionType destinationDimension, EntityPlacer placementLogic) {
+        if(entity.getEntityWorld().isClient) {
+            System.out.println("ONLY CALL FROM SERVER");
```

Comment:
A logger would be much better.

Reply:
A crash would probably be best since that will make it harder to just ignore the error.

Reply:
Yes, hard crash here, if something is up make is obvious